mgechev/angular-seed
# feat(build): Another stab at having rollup tree-shaking in our build.

To test this out, run:

`npm run serve.prod.rollup.exp`

Which will perform a regular prod build, but also tree-shake the app.js bundle with rollup. 

It will not tree-shake rxjs, but it shouldn't really matter given that only used operators should be imported, and the core content (Observables, Subject, etc) are used by most operators and apps. 

This is largely based on the work done in the rollup branch, but I did a few small changes like abandoning the idea of using rxjs-es (given that it is no longer published) and adding the commonsjs rollup support.

As of right now, the sample app that comes with the seed can be built without any changes or workarounds.

Preliminary results for the size of the generated app.js bundles look like this:

**serve.prod**
   app.js size: 1MB

**serve.prod.exp**
   app.js size: 623 Kb

**serve.prod.rollup.exp**
   app.js size: 426 Kb


What do you guys think?
